### PR TITLE
ci(next): fix Drupal version constraints in Composer commands

### LINKED ISSUE #545: CI uses fixed 10.x.0 Drupal version instead of 10.x.x
In the `next.yml` GitHub CI file, the following commands are run:

```cli
composer create-project drupal/recommended-project:10.0 ~/drupal --no-interaction --no-install
composer require drupal/core-dev:10.0 --no-suggest --no-install
```

But the syntax `project:10.0` tells composer to use the exact `10.0.0` version. 

But the CI should definitely be using the latest `10.0.x`. The composer dependency syntax for that is `project:~10.0.0` or `project:10.0.x` which says to use any 10.0.x version below 10.1.0.

For more details, see the [composer docs for version constraints](https://getcomposer.org/doc/articles/versions.md#wildcard-version-range-)

Also the version number on the `drupal/core-dev` dependency is not needed because Composer will resolve it to a version that matches the `drupal/recommend-project` version.

Here's the CLI output of running the existing commands:
```cli
% composer create-project drupal/recommended-project:10.0 ~/drupal --no-interaction --no-install
Creating a "drupal/recommended-project:10.0" project at "~/drupal"
Info from https://repo.packagist.org:
Installing drupal/recommended-project (10.0.0)
  - Installing drupal/recommended-project (10.0.0): Extracting archive
Created project in ~/drupal

% cd ~/drupal
% composer require drupal/core-dev:10.0 --no-suggest --no-install
You are using the deprecated option "--no-suggest". It has no effect and will break in Composer 3.
./composer.json has been updated
Running composer update drupal/core-dev
Loading composer repositories with package information
Updating dependencies
Lock file operations: 73 installs, 0 updates, 0 removals
  - Locking drupal/core-dev (10.0.0)
```

Note that it is using Drupal 10.0.0 even though 10.0.11 is available. Also, not the warning about the `--no-suggest` option too.

And here's the output of Composer using the proper version syntax:

```cli
% composer create-project drupal/recommended-project:10.0.x ~/drupal --no-interaction --no-install
Creating a "drupal/recommended-project:10.0.x" project at "~/drupal"
Info from https://repo.packagist.org:
Installing drupal/recommended-project (10.0.11)
  - Installing drupal/recommended-project (10.0.11): Extracting archive
Created project in ~/drupal

% composer require drupal/core-dev --no-install
./composer.json has been updated
Running composer update drupal/core-dev
Loading composer repositories with package information
Updating dependencies
Lock file operations: 74 installs, 0 updates, 0 removals
  - Locking drupal/core-dev (10.0.11)
```

Note that it does _not_ use Drupal 10.1.5 even though that is the latest version of Drupal. It sticks to the 10.0.x range we wanted.

### PR DESCRIPTION
**Pull request by JohnAlbin**
Fixes #545

### CODE REVIEW

File: .github/workflows/next.yml [hunk 1]
```diff
@@ -56,8 +56,7 @@ jobs:
           composer config preferred-install dist
           composer config repositories.0 path $GITHUB_WORKSPACE
           composer config repositories.1 composer https://packages.drupal.org/8
-          composer config allow-plugins true -n
```
Comment (robdecker) [inline .github/workflows/next.yml:59]: By removing the `allow-plugins` command, this could cause the CI to hang at a prompt, right?

Comment (robdecker) [reply .github/workflows/next.yml:59]: Looking at a CI build log, it doesn't look like there are any plugins. I suppose there could be some dependency in the future that requires a plugin that would need to be allowed to run? I guess the composer `--no-interaction` flag would keep the build from hanging. But then the plugin wouldn't run...

Comment (JohnAlbin) [reply .github/workflows/next.yml:59]: Ah! It prevents a command prompt when installing a Composer plugin in the CI. I didn't realize that that is what the purpose of the command is. I've removed the removal.

**Review by robdecker — APPROVED** [review]
LGTM 👍

**Pull request merged**